x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension/endpoint_policy_create_extension.tsx

```diff
@@ -0,0 +1,354 @@
+import type { PackagePolicyCreateExtensionComponentProps } from '@kbn/fleet-plugin/public';
+import { useLicense } from '../../../../../../common/hooks/use_license';
+import {
+  ALL_EVENTS,
+  CLOUD_SECURITY,
+  EDR_COMPLETE,
+  NGAV,
+  EDR_ESSENTIAL,
+  ENDPOINT,
+  INTERACTIVE_ONLY,
+  PREVENT_MALICIOUS_BEHAVIOUR,
+} from './translations';
+
+const PREFIX = 'endpoint_policy_create_extension';
+
+const ENDPOINT_INTEGRATION_CONFIG_KEY = 'ENDPOINT_INTEGRATION_CONFIG';
+
+const environmentMapping = {
+  cloud: CLOUD_SECURITY,
+  endpoint: ENDPOINT,
+};
+
+const endpointPresetsMapping = {
+  NGAV,
+  EDREssential: EDR_ESSENTIAL,
+  EDRComplete: EDR_COMPLETE,
+};
+
+const cloudEventMapping = {
+  INTERACTIVE_ONLY,
+  ALL_EVENTS,
+};
+
+type EndpointPreset = keyof typeof endpointPresetsMapping;
+type CloudEvent = keyof typeof cloudEventMapping;
+type Environment = keyof typeof environmentMapping;
+
+const environmentOptions: Array<{ value: Environment; text: string }> = [
+  { value: 'endpoint', text: ENDPOINT },
+  { value: 'cloud', text: CLOUD_SECURITY },
+];
+
+const HelpTextWithPadding = styled.div`
+  padding-left: ${(props) => props.theme.eui.euiSizeL};
+`;
+
+/**
+ * Exports Endpoint-specific package policy instructions
+ * for use in the Ingest app create / edit package policy
+ */
+export const EndpointPolicyCreateExtension = memo<PackagePolicyCreateExtensionComponentProps>(
+  ({ newPolicy, onChange }) => {
+    const isPlatinumPlus = useLicense().isPlatinumPlus();
+
+    // / Endpoint Radio Options (NGAV and EDRs)
+    const [endpointPreset, setEndpointPreset] = useState<EndpointPreset>('NGAV');
+    const [behaviorProtectionChecked, setBehaviorProtectionChecked] = useState(false);
+    const [selectedCloudEvent, setSelectedCloudEvent] = useState<CloudEvent>('INTERACTIVE_ONLY');
+    const [selectedEnvironment, setSelectedEnvironment] = useState<Environment>('endpoint');
+    const initialRender = useRef(true);
+
+    // Fleet will initialize the create form with a default name for the integratin policy, however,
+    // for endpoint security, we want the user to explicitely type in a name, so we blank it out
+    // only during 1st component render (thus why the eslint disabled rule below).
+    // Default values for config are endpoint + NGAV
+    useEffect(() => {
+      onChange({
+        isValid: false,
+        updatedPolicy: {
+          ...newPolicy,
+          name: '',
+          inputs: [
+            {
+              enabled: true,
+              streams: [],
+              type: ENDPOINT_INTEGRATION_CONFIG_KEY,
+              config: {
+                _config: {
+                  value: {
+                    type: 'endpoint',
+                    endpointConfig: {
+                      preset: 'NGAV',
+                    },
+                  },
+                },
+              },
+            },
+          ],
+        },
+      });
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can we avoid this here and below?

Excluding dependencies from a memoized utility can lead to bugs that are hard to debug.

Reply:
hey @paul-tavares I managed to remove the ignore depency error by adding a custom comparator to the memo, can you take a look again?